### PR TITLE
Add asio as a submodule

### PR DESCRIPTION
**Pull request by jroweboy**
Fixes #Subv pesters random people on IRC to get them to make this extremely minor change instead of writing the 2 lines of code.

As far as I know Subv wants asio added so that soc:u and maybe some other things can use that instead.

---
This change is [Reviewable](https://reviewable.io/reviews/citra-emu/citra/2563)

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -271,6 +271,9 @@ endif()
 # process subdirectories
 add_subdirectory(externals/soundtouch)
 
+add_definitions(-DASIO_STANDALONE=1)
+include_directories(externals/asio/asio/include)
```
Comment (yuriks) [inline CMakeLists.txt]: Add this to the CMakeFiles in externals/ instead.

Comment (wwylele) [inline CMakeLists.txt]: I think this should be added to the one in the subdirectory that uses asio? (In this case, probably  `src/core/CMakeLists.txt`). Or just remove this and let the one add this who are going to write code using asio

Comment (yuriks) [inline CMakeLists.txt]: For some reason I thought that ASIO had a CMake file that we could just import.
The proper way of doing it is using the CMake support for library interfaces. See what I did for xbyak for example.

**Comment by Saphiresurf** [comment]
This will be nice!

**Comment by jfmherokiller** [comment]
i think this change would be better on the https://github.com/citra-emu/ext-boost submodule since it is built into boost if added by mcp.

**Comment by MerryMage** [comment]
@jfmherokiller Doing so (minorly) complicates the build, because the boost version of asio requires boost::system.

**Comment by jfmherokiller** [comment]
oh i see and boost system is already included in c++11 so it would add duplicates.

**Comment by yuriks** [comment]
We'd need to create a CMake-based build system for the boost version anyway. The standalone version is updated more often and I'd prefer using as many native C++ features as possible instead of the older boost alternatives.

**Comment by yuriks** [comment]
Just to be sure, can you actually link asio into one of our libraries with this target? (Just to prevent a following PR to fix it if that ends up failing.) If so, then LGTM

**Comment by bunnei** [comment]
@Subv do we still want this? I don't see the point of merging this if we're not using it. Do you plan on making the changes to use it in the near term? Otherwise, this should be closed and re-added when we are ready for it.

**Comment by mailwl** [comment]
+1 for this, it's "стильно, модно, молодёжно". Sorry, don't know how to proper translate this meme to english

**Comment by Subv** [comment]
That depends on how you define "near term", i plan on rewriting the soc module with it, but that's blocked until a few more of my kernel and threading patches make it in.

**Comment by mailwl** [comment]
yes, Youtube app still not working

**Comment by jroweboy** [comment]
As this is really old and not necessary at this moment, I'm just going to close this and if someone wants to use asio they can add it themselves (its really straightforward).

Also @yuriks I forgot to mention, but I did have a commit after this one on my local copy that tests asio to make sure it works.

**Pull request closed without merging**